eZ/Publish/Core/MVC/Symfony/Routing/UrlAliasRouter.php

```diff
@@ -296,6 +296,14 @@ public function getRouteCollection()
      */
     public function generate($name, $parameters = [], $referenceType = UrlGeneratorInterface::ABSOLUTE_PATH)
     {
+        if ($name === '' &&
+            array_key_exists(RouteObjectInterface::ROUTE_OBJECT, $parameters) &&
+            $this->supportsObject($parameters[RouteObjectInterface::ROUTE_OBJECT])
+        ) {
+            $name = $parameters[RouteObjectInterface::ROUTE_OBJECT];
+            unset($parameters[RouteObjectInterface::ROUTE_OBJECT]);
+        }
+
         // Direct access to Location
         if ($name instanceof Location) {
```

Comment:
Correct me if I'm wrong, but due to change in RoutingExtension:
```php
        if (is_object($name)) {
            return $this->generateUrlForObject($name, $parameters, $referenceType);
        }
```
you no longer need that, right?

Probably needs to go anyway due to my previous comment.